base/abstractset.jl

```diff
@@ -251,17 +251,16 @@ true
 issubset, ⊆, ⊇
 
 function issubset(l, r)
-    if haslength(r)
-        rlen = length(r)
+    if haslength(r) && (isa(l, AbstractSet) || !hasfastin(r))
+        rlen = length(r) # conditions above make this length computed only when needed
         if isa(l, AbstractSet)
             # check l for too many unique elements
             length(l) > rlen && return false
         end
-        # if r is big enough, convert it to a Set
-        # threshold empirically determined by repeatedly
-        # sampling using these two methods (see #26198)
-        if rlen > 70 && !isa(r, AbstractSet)
-            return issubset(l, Set(r))
+        if !hasfastin(r)
+            # if r is big enough, convert it to a Set
+            # this threshold was empirically determined (cf. #26198)
+            rlen > 70 && return issubset(l, Set(r))
```

Comment:
Why not put this in the `if` line above?

Reply:
I would say for the same reason as the previous check with `if isa(l, AbstractSet)`, to allow commenting between the comments. But I kinda agree that two nested simple `if` should be combined, I will update.